pilot/pkg/bootstrap/server.go

```diff
@@ -690,19 +639,31 @@ func (s *Server) initEventHandlers() error {
 }
 
 // add a GRPC listener using DNS-based certificates. Will be used for Galley, injection and CA signing.
-func (s *Server) initDNSListener() error {
-	if features.IstiodService.Get() == "" || s.kubeClient == nil {
+func (s *Server) initDNSListener(args *PilotArgs) error {
+	istiodAddr := features.IstiodService.Get()
+	if istiodAddr == "" || s.kubeClient == nil {
 		// Feature disabled
 		return nil
 	}
+
+	// validate
+	host, port, err := net.SplitHostPort(istiodAddr)
+	if err != nil {
+		return fmt.Errorf("invalid ISTIOD_ADDR(%s): %v", istiodAddr, err)
+	}
+	_, err = strconv.Atoi(port)
```

Comment:
This might be cleaner as: 

```
if  _, err = strconv.Atoi(port); err != nil {
	return fmt.Errorf("invalid ISTIOD_ADDR(%s): %v", istiodAddr, err)
}
```

Reply:
make sense